datashader/spatial/__init__.py

```diff
@@ -0,0 +1,621 @@
+                            x,
+                            y,
+                            filename,
+                            p=10,
+                            npartitions=None,
+                            shuffle=None,
+                            compression='default'):
+        """
+        Perform spatial partitioning on an input dataframe and write the
+        result to a parquet file.  The resulting parquet file will contain
+        the same columns as the input dataframe, but the dataframe's original
+        index will be dropped.
+
+        The resulting parquet file will contain all of the rows from the
+        input dataframe, but they will be spatially sorted and partitioned
+        along a 2D Hilbert curve (https://en.wikipedia.org/wiki/Hilbert_curve).
+
+        The parquet file will also contain custom metadata that is needed to
+        reconstruct the Hilbert curve distances on load.  This parquet file
+        may then be used to construct SpatialPointsFrame instances.
+
+        Parameters
+        ----------
+        df: pd.DataFrame or dd.DataFrame
+            The input dataframe to partition
+        x, y
+            The column labels in df of the x and y coordinates of each row
+        filename: str
+            The path where the resulting parquet file should be written.
+            See dask.dataframe.to_parquet for description of supported path
+            specifications.
+        p: int (default 10)
+            The Hilbert curve order parameter that determines the resolution
+            of the 2D grid that data points are rounded to before computing
+            their Hilbert distance. Points will be discretized into 2 ** p
+            bins in each the x and y dimensions.
+
+            This parameter should be increased if the partitions of the
+            resulting parquet files are significantly unbalanced.
+
+        npartitions: int or None (default None)
+            The number of partitions for the resulting parquet file.  If None
+            (the default) this is chosen to be the greater of 8 and
+            len(df) // 2**23.
+
+            In general, increasing the number of partitions will improve
+            performance when processing small subsets of the overall parquet
+            data set.  But this comes at the cost of some additional overhead
+            when processing the entire data set.
+
+        shuffle: str or None (default None)
+            The dask.dataframe.DataFrame.set_index shuffle method. If None,
+            a default is chosen based on the current scheduler.
+
+        compression: str or None (default)
+            The dask.dataframe.to_parquet compression method.
+        """
+
+        _validate_fastparquet()
+
+        # Validate filename
+        if (not isinstance(filename, basestring) or
+                not filename.endswith('.parquet')):
+            raise ValueError(
+                'filename must be a string ending with a .parquet extension')
```

Comment:
We've been using .parq for the extension, so please allow that convention as well.

Reply:
done in ff82fe1